samcli/cli/cli_config_file.py

```diff
@@ -76,7 +75,6 @@ def __call__(self, config_path, config_env, cmd_names):
 
             # NOTE(TheSriram): change from tomlkit table type to normal dictionary,
             # so that click defaults work out of the box.
-            samconfig.sanity_check()
```

Comment:
What was this doing here? We don't need it anymore?

Reply:
Right, I explain the why in the summary. Basically `get_all()` already does what `sanity_check()` does, except that `sanity_check()` swallows the important TOML exception with the useful error message.